bleak/backends/corebluetooth/scanner.py

```diff
@@ -40,11 +39,65 @@ def __init__(self, **kwargs):
     async def start(self):
         self._identifiers = {}
 
+        def safe_list_get(l: list, idx: int, default):
+            """
+            Returns an index from a list safely similar to .get() with dicts
+            """
+            try:
+                return repr(l[idx]).lower()
+            except IndexError:
+                return default
+
+        def callback_dict_breakdown(data) -> Union[None, dict]:
+            """
+            This function parses the __NSDictionaryM or __NSSingleEntryDictionry
+            object passed as data into a more user-friendly dictionary which
+            is returned, if data is None, or if an exception is raised during
+            parsing, None is returned
+            """
+            if data:
+                try:
+                    _service_dict_key = data.allKeys()[0]
+                    _service_dict_data = data.objectForKey_(_service_dict_key)
+                    return {str(_service_dict_key): str(_service_dict_data)}
+                except Exception:
+                    return None
+            else:
+                return None
+
         def callback(p, a, r):
             # update identifiers for scanned device
             self._identifiers.setdefault(p.identifier(), {}).update(a)
-            if self._callback:
-                self._callback(p, a, r)
+
+            # Process service data
+            service_data_dict_raw = a.get("kCBAdvDataServiceData", {})
+            service_data = {
+                cb_uuid_to_str(k): bytes(v) for k, v in service_data_dict_raw.items()
+            }
+
+            # Process manufacturer data into a more friendly format
+            manufacturer_binary_data = a.get("kCBAdvDataManufacturerData")
+            manufacturer_data = {}
+            if manufacturer_binary_data:
+                manufacturer_id = int.from_bytes(
+                    manufacturer_binary_data[0:2], byteorder="little"
+                )
+                manufacturer_value = bytes(manufacturer_binary_data[2:])
+                manufacturer_data = {manufacturer_id: manufacturer_value}
```

Comment:
This could modify the existing dictionary instead of creating a new one

    manufacturer_data[manufacturer_id] = manufacturer_value